### ISSUE TITLE
[Bug]:user have table level priv select but internal error: don‘t have priv to execute statement 

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#5214]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

SQL：
create role if not exists low_level,mid_level,high_level;
grant select,insert on table testdb.* to low_level;
create database testdb;
create table testdb.aaa(b int);
insert into testdb.aaa values(1);
create user usr_role identified by '12345678' default role low_level;
grant select,insert on table testdb.* to low_level;
grant low_level to usr_role;
mysql -h 127.0.0.1 -P 6001 -u sys:usr_role -p
select * from testdb.aaa;
![image](https://user-images.githubusercontent.com/113406637/191879577-9dfef814-a39a-449b-ab6c-e54419d28e39.png)
![image](https://user-images.githubusercontent.com/113406637/191879591-e5d14c2c-0456-42b3-991e-f54c93a834f5.png)

**Comment by daviszhen** [comment]
it was fixed at #5411 .
But, `create database testdb;` must be executed first  to ensure the database testdb exists .
Then, execute `grant select,insert on table testdb.* to low_level;`.
The Grant statement will check the existence of the testdb.

**Comment by daviszhen** [comment]
create role if not exists low_level,mid_level,high_level;
create database testdb;
grant select,insert on table testdb.* to low_level;
create table testdb.aaa(b int);
insert into testdb.aaa values(1);
